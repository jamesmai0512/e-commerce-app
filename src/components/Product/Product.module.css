.product_item {
  display: flex;
  flex-direction: column;
  align-items: start;
  flex: 1 0 calc(25% - 10px);
  max-width: 320px;
  cursor: pointer;
}

.product_title {
  font-size: 20px;
  padding-top: 20px;
  line-height: 27px;
}

.product_price {
  font-size: 18px;
  line-height: 27px;
}

.small {
  width: 305px;
  height: 375px;
  flex: 1;
}

.product {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  padding: 50px 80px;
  cursor: pointer;
  font-family: 'ClashDisplay Variable', sans-serif;
}

.product_layout {
  position: relative;
}

.main {
  padding-bottom: 380px;
}

.product_details {
  display: flex;
  flex-direction: column;
  align-items: start;
  color: #2a254b;
  font-family: 'ClashDisplay Variable', sans-serif;
  padding-bottom: 40px;
}

.large {
  width: 630px;
  height: 375px;
  flex: 2;
  cursor: pointer;
}

.product_detail {
  display: flex;
}

.product_detail_content {
  padding: 95px;
  color: #2a254b;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.product_detail_content_title {
  font-size: 36px;
  line-height: 44.28px;
}

.product_detail_content_price {
  font-size: 24px;
  line-height: 32.4px;
}

.product_detail_content_description {
  font-size: 16px;
  line-height: 20px;
}

.product_detail_content_description_content {
  color: #505977;
  font-size: 16px;
  line-height: 20px;
}

.product_detail_content_description_list_content {
  padding-top: 20px;
  padding-left: 10px;
}

.product_detail_content_description_list_content li::before {
  content: 'â€¢   ';
}

.cart_function_item {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.input_amount {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 22px;
}

.product_detail_content_dimensions {
  display: flex;
  gap: 58px;
}

.product_detail_content_dimensions_size {
  line-height: 30px;
}

.introduce_brand {
  display: flex;
  justify-content: center;
  padding: 60px 0 50px;
  color: #2a254b;
  font-size: 24px;
}

.order_step {
  display: flex;
  padding: 90px 100px;
  gap: 58px;
}

.email_form {
  background: #f9f9f9;
  padding: 52px 65px;
}

.form_text {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 16px;
  padding: 68px 45px;
  background: white;
}

.form_text_title {
  font-size: 36px;
  display: flex;
  justify-content: center;
}

.form_email_submit {
  display: flex;
  width: 440px;
}

.form_text_description {
  padding-bottom: 72px;
}
