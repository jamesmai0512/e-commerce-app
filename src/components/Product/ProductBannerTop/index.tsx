import Image, { StaticImageData } from 'next/image'
import { memo } from 'react'

// Components
import Button from '../../Button'

// Styles
import styles from './ProductBannerTop.module.css'

// Types
export type ProductBannerTopProps = {
  title: string
  buttonTheme: string
  buttonText: string
  description: string
  imageUrl: string
  onButtonClick?: (
    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
  ) => void
}

const ProductBannerTop = ({
  title,
  buttonTheme,
  buttonText,
  description,
  imageUrl,
  onButtonClick,
}: ProductBannerTopProps) => {
  const {
    product_banner,
    product_banner_content,
    product_banner_title,
    product_banner_description,
  } = styles
  return (
    <div className={product_banner}>
      <div className={product_banner_content}>
        <div>
          <h2 className={product_banner_title}>{title}</h2>
          <Button
            onClick={onButtonClick}
            background={buttonTheme}
            text={buttonText}
          />
        </div>

        <p className={product_banner_description}>{description}</p>
      </div>
      <div className="product_banner_image">
        <Image
          src={imageUrl}
          alt="product-banner"
          loading="eager"
          width={520}
          height={584}
        />
      </div>
    </div>
  )
}

export default memo(ProductBannerTop)
