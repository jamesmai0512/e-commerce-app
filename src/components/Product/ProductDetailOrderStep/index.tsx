import dynamic from 'next/dynamic'

// Component
const BoxInfo = dynamic(() => import('@/components/Box/BoxInfo'))

// Types
import { imageRoute, ORDERSTEPS, TProduct } from '@/constants/common'

// Image
const BigChair = imageRoute('BigChair')

// Styles
import styles from './Product.module.css'

interface Props {
  product: TProduct
}

const ProductDetailOrderStep = ({ product }: Props) => {
  const { order_step } = styles

  return (
    <div className={order_step}>
      {ORDERSTEPS.map((i) => (
        <BoxInfo
          key={i.id}
          icon={i.icon}
          title={i.title}
          description={i.description}
        />
      ))}
    </div>
  )
}
export default ProductDetailOrderStep
