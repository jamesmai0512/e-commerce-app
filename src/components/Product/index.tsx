import { memo } from 'react'
import { useRouter } from 'next/router'
import Image, { StaticImageData } from 'next/image'

// Styles
import styles from './Product.module.css'

export type ProductProps = {
  imageUrl: string
  title: string
  price: number
  imageSize: string
  id?: number
}

const Product = ({ id, imageUrl, title, price, imageSize }: ProductProps) => {
  const router = useRouter()

  const {
    product_item,
    product_image,
    product_details,
    product_title,
    product_price,
  } = styles

  const handleClickProduct = (
    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
  ) => {
    event.preventDefault()
    router.push(`/products/${id}`)
  }

  return (
    <div
      onClick={handleClickProduct}
      className={imageSize === 'small' ? `${product_item}` : ''}
    >
      <div className={`${product_image} ${styles[imageSize]}`}>
        <Image
          src={imageUrl}
          alt="product-image"
          width={imageSize === 'small' ? '305' : '630'}
          height="375"
          loading="eager"
        />
      </div>
      <div className={product_details}>
        <h3 className={product_title}>{title}</h3>
        <p className={product_price}>${price}</p>
      </div>
    </div>
  )
}

export default memo(Product)
