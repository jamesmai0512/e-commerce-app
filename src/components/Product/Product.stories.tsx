import React from 'react'
import { Meta } from '@storybook/react'
import Product, { ProductProps } from './'

const mockProduct: ProductProps = {
  imageUrl:
    'https://raw.githubusercontent.com/jamesmai0512/image-ecommerce/main/images/Lamp.png',
  title: 'Mock Product',
  price: 9.99,
  imageSize: 'small',
}

export default {
  title: 'Components/Product',
  component: Product,
} as Meta

export const Default = () => <Product {...mockProduct} />
