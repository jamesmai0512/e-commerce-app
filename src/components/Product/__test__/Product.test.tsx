import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import { useRouter } from 'next/router'
import Product, { ProductProps } from '../'

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}))

describe('Product', () => {
  const mockRouterPush = jest.fn()
  const props: ProductProps = {
    imageUrl: 'https://example.com/image.jpg',
    title: 'Test Product',
    price: 9.99,
    imageSize: 'small',
    id: 123,
  }

  beforeEach(() => {
    jest.clearAllMocks()
    ;(useRouter as jest.Mock).mockReturnValue({
      push: mockRouterPush,
    })
  })

  it('Should renders the product correctly', () => {
    const component = render(<Product {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('Should renders the product with the correct props', () => {
    const { getByText, getByAltText } = render(<Product {...props} />)
    expect(getByAltText('product-image')).toBeInTheDocument()
    expect(getByText('Test Product')).toBeInTheDocument()
    expect(getByText('$9.99')).toBeInTheDocument()
  })
})
