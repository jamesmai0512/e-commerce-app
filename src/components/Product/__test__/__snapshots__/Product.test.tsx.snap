// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Should renders the product correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="product_item"
      >
        <div
          class="product_image small"
        >
          <img
            alt="product-image"
            data-nimg="1"
            decoding="async"
            height="375"
            loading="eager"
            src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="305"
          />
        </div>
        <div
          class="product_details"
        >
          <h3
            class="product_title"
          >
            Test Product
          </h3>
          <p
            class="product_price"
          >
            $
            9.99
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="product_item"
    >
      <div
        class="product_image small"
      >
        <img
          alt="product-image"
          data-nimg="1"
          decoding="async"
          height="375"
          loading="eager"
          src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="305"
        />
      </div>
      <div
        class="product_details"
      >
        <h3
          class="product_title"
        >
          Test Product
        </h3>
        <p
          class="product_price"
        >
          $
          9.99
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
